### PR TITLE
skip conda-python-tests on arm64

### PR DESCRIPTION
**Pull request by jameslamb**
All of this repo's `conda-python-tests` jobs have conditions in them like "skip on ARM":

https://github.com/rapidsai/cugraph-gnn/blob/2dd300122dfd6fdea70c9d20c276a3c5946b7613/ci/test_python.sh#L100

https://github.com/rapidsai/cugraph-gnn/blob/2dd300122dfd6fdea70c9d20c276a3c5946b7613/ci/test_python.sh#L141

https://github.com/rapidsai/cugraph-gnn/blob/2dd300122dfd6fdea70c9d20c276a3c5946b7613/ci/test_python.sh#L183

As a result, right now the arm64 `conda-python-tests` jobs are just wasting CI resources... they're spending ~40+~ 5-10 minutes occupying a GPU runner just to download some datasets and then exit ([example build link](https://github.com/rapidsai/cugraph-gnn/actions/runs/11858773988/job/33056063652?pr=69)).

This proposes never even starting those jobs, to make CI here less expensive.

### But why are we skipping arm at all?

Lack of pytorch packages. See https://github.com/rapidsai/cugraph-gnn/pull/61#discussion_r1822773547

### CODE REVIEW

**Comment by jameslamb** [comment]
Looks like this didn't get in last week. It's not critical, so I've re-targeted it at `branch-25.02`.

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Comment by jameslamb** [comment]
This won't be mergeable until we get a `ci-codeowners` approval. @AyodeAwe could you take a look here? (the bot picked you for this one)

**Review by AyodeAwe — APPROVED** [review]

**Comment by jameslamb** [comment]
Thanks!

**Pull request merged**
